### PR TITLE
Fixed #23051 -- Raise exception on only() calls with reverse fields

### PR DESCRIPTION
**Pull request by czpython**
Fixed https://code.djangoproject.com/ticket/23051

### CODE REVIEW

**Comment by czpython** [comment]
Test failure appears to be "random" 🤔

File: django/db/models/query.py [hunk 1]
```diff
@@ -1168,10 +1168,24 @@ def only(self, *fields):
             # Can only pass None to defer(), not only(), as the rest option.
             # That won't stop people trying to do this, so let's be explicit.
             raise TypeError("Cannot pass None as an argument to only().")
+
+        reverse_fields = set(
```
Comment (jdufresne) [inline django/db/models/query.py]: This can use set comprehension

 ```suggestion
         reverse_fields = {
```

**Comment by nessita** [comment]
@czpython Hi! I'm doing some PR cleanup, would you have time to update this PR with the latest main? There are also some questions for you to answer in the associated ticket. Thank you!

**Comment by nessita** [comment]
Closing due to inactivity, feel free to re-open if you can continue working on this. Thanks!

**Pull request closed without merging**